### PR TITLE
Add LargeDialog component and story

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Animated modal, full-screen on mobile, large and centered on desktop

<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/GLY3tSM85RFBvAf6lFNA/b36a3a76-0742-48a9-9f7f-f96dd29cf667.mov">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/GLY3tSM85RFBvAf6lFNA/b36a3a76-0742-48a9-9f7f-f96dd29cf667.mov">Screen Recording 2024-10-18 at 13.43.11.mov</video>

## Justify why they are needed

Better than current `FullscreenModal` for most use cases

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
* **#4872** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4872?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/ui/src/components/Dialog/LargeDialog.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { type Meta, type StoryFn } from '@storybook/react'
+import { yStack } from '../../patterns'
+import { Button } from '../Button/Button'
+import { Text } from '../Text/Text'
+import * as LargeDialog from './LargeDialog'
```
Comment (Youakeem) [inline packages/ui/src/components/Dialog/LargeDialog.stories.tsx:5]: Nit: It's easier with auto imports if we expose an object instead individual pieces

```
export const LargeDialog = {
  Root: ...,
  Header: ...,
  Trigger: ...,
  Content: ...,
}
```

**Review by Youakeem — APPROVED** [review]

File: packages/ui/src/components/Dialog/LargeDialog.stories.tsx (hunk 1 above)
Comment (alebedev) [reply packages/ui/src/components/Dialog/LargeDialog.stories.tsx:5]: Yes, this is much better!

File: packages/ui/src/components/Dialog/LargeDialog.css.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+export const content = style({
+  display: 'flex',
+  width: '100%',
+})
+
+export const card = style({
+  position: 'relative',
+  marginBlock: 'auto',
+  marginInline: tokens.space.md,
```
Comment (gustaveen) [inline packages/ui/src/components/Dialog/LargeDialog.css.ts:18]: it seem to have 24px padding in [Figma](https://www.figma.com/design/p1TEzLLVqJB9gKDNTDVGU4/Hedvig.com?node-id=3400-127452&m=dev)
```suggestion
  marginInline: tokens.space.lg,
```

File: packages/ui/src/components/Dialog/LargeDialog.css.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { style } from '@vanilla-extract/css'
+import { minWidth, tokens } from '../../theme'
+
+export const closeButton = style({
+  position: 'absolute',
+  right: tokens.space.md,
+  top: tokens.space.md,
```
Comment (gustaveen) [inline packages/ui/src/components/Dialog/LargeDialog.css.ts:7]: Update with if updating padding, see comment below
```suggestion
  right: tokens.space.lg,
  top: tokens.space.lg,
```

File: packages/ui/src/components/Dialog/LargeDialog.css.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  // If card has header inside, then other content should generally have its own wrapper with overflowY: 'auto'
+  // to make header and close button sticky. This is a fallback for dialogs which don't have a header
+  overflowY: 'auto',
+
+  '@media': {
+    [minWidth.sm]: {
+      marginBlock: 'auto',
+      marginInline: 'auto',
+      maxWidth: 'min(60rem, 80%)',
+    },
```
Comment (gustaveen) [inline packages/ui/src/components/Dialog/LargeDialog.css.ts:30]: It has larger padding-right in desktop to make room for close button
```suggestion
    },
    [minWidth.lg]: {
      paddingRight: tokens.space.xxl',
    },
```

File: packages/ui/src/components/Dialog/LargeDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import { type ComponentProps, type HTMLProps, type ReactNode } from 'react'
+import { CrossIcon } from '../../icons'
+import { framerTransitions } from '../../theme'
+import { IconButton } from '../Button/IconButton'
+import { Card } from '../Card/Card'
+import { Heading } from '../Heading/Heading'
+import * as Dialog from './Dialog'
+import { card, closeButton, content, heading, scrollableInnerContent } from './LargeDialog.css'
+
+// Large modal, full-screen on mobile
```
Comment (gustaveen) [inline packages/ui/src/components/Dialog/LargeDialog.tsx:12]: nit: maybe add a comment that it uses Root and Trigger components from regular Dialog?

**Review by gustaveen — APPROVED** [review]
Looks good! 🙌 Just saw some spacing tweaks and the background blur for the dialog overlay is different in new design, 100px compared to 64px. Up to you if you want to tweak it in a separate PR

File: packages/ui/src/components/Dialog/LargeDialog.css.ts (hunk 4 above)
Comment (alebedev) [reply packages/ui/src/components/Dialog/LargeDialog.css.ts:30]: I assumed only header need padding. This has its pros and cons, will talk to Nils before making a decision

File: packages/ui/src/components/Dialog/LargeDialog.css.ts (hunk 2 above)
Comment (alebedev) [reply packages/ui/src/components/Dialog/LargeDialog.css.ts:18]: This represents outer margin. I'll make sure inner padding is 1.5rem

**Review by guilhermespopolin — APPROVED** [review]

File: packages/ui/src/components/Dialog/LargeDialog.css.ts (hunk 4 above)
Comment (alebedev) [reply packages/ui/src/components/Dialog/LargeDialog.css.ts:30]: Nils approved having padding just in the header, it looks better overall even if header-less content may in rare cases go under the button

**Comment by alebedev** [comment]
### Merge activity

* **Oct 24, 3:18 AM EDT**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4872).

**Pull request merged**